features/step_definitions/team_steps.rb

```diff
@@ -32,3 +34,21 @@
 Then(/^I click on team title$/) do
   find('#team-name').click
 end
+
+Given('the following users are registered with teams:') do |table|
+  table.hashes.each do |row|
+    team = Team.find_by(name: row[:team])
+    team ||= FactoryBot.create(:team, name: row[:team])
+    user = FactoryBot.create(:user, row.slice('name', 'email'))
+    FactoryBot.create(:user_team, user: user, team: team, role: row[:role])
+  end
+end
+
+Given('team settings page') do
+  visit teams_path
+  page.save_screenshot(full: true)
+end
+
+Then('I click leave team button') do
+  find("a[data-action='leave-user-team']").click
```

Comment:
Better use text identifier and you can use it as general element:
```
Then("I click {string} button") do |button|
  find('.btn', text: button).click
end
```

And then call it 
```
And I click "Leave Team" button
```

Reply:
The button contains 2 spans with an icon, is not text-only btn.